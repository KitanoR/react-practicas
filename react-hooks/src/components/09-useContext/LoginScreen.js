import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { user, setUser } = useContext(UserContext);
    return (
        <div>
            <h1>Login Screen</h1>
            <hr />
            <button
                className="btn btn-primary"
                onClick={() => setUser({ id: 1, name: 'cayetano', age: 23 })}
            >
                Login
            </button>
        </div>
    )
}
