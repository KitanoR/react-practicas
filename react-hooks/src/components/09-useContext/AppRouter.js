import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from "react-router-dom";
import {AboutScreen} from './AboutScreen';
import {LoginScreen}  from './LoginScreen';
import {HomeScreen} from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className="container">
                    <Switch>
                        <Route path="/about" exact component={AboutScreen} />
                        <Route path="/login" exact component={LoginScreen} />
                        <Route path="/" exact component={HomeScreen} />
                        <Redirect to="/" />
                    </Switch>
                </div>
                
            </div>
        </Router>
    )
}
