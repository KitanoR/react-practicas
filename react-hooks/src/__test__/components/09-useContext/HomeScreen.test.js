import React  from 'react';

import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { mount } from 'enzyme';


describe('Pruebas de home screen', () => {
    const user = {
        name: 'caye',
        email: 'caye@yaho.com'
    }
    // Mount se utilia para renderizar todo en general
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    )
    test('should debe mostrarse ocrrectamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
