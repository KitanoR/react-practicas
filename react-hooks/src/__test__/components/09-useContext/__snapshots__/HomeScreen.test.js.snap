// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de home screen should debe mostrarse ocrrectamente 1`] = `
<div>
  <h1>
    Home screen
  </h1>
  <hr />
  <pre>
    {
   "name": "caye",
   "email": "caye@yaho.com"
}
  </pre>
</div>
`;
